__tests__/app.test.js

```diff
@@ -137,6 +137,36 @@ describe("/api/articles", () => {
         expect(response.body.msg).toBe("Invalid order");
       });
   });
+
+  test("GET: 200 - filters articles by topic", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toHaveLength(12);
+        response.body.articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("GET: 404 - responds with error if topic does not exist", () => {
+    return request(app)
+      .get("/api/articles?topic=bad_topic")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "Topic not found or no articles for this topic"
+        );
+      });
+  });
+  test("GET: 200 - returns all articles if no topic is provided", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles.length).toBe(13);
+      });
+  });
```

Comment:
There's a single topic that does exist but has no articles attached to it yet, you could test for this and make sure that it returns an empty array and 200